src/docker-osx-dev

```diff
@@ -375,17 +381,74 @@ function init_boot2docker {
     log_info "Starting Boot2Docker VM"
     boot2docker start --vbox-share=disable
   fi
+
+  configure_boot2docker
+}
+
+################################################################################
+# docker-machine manipulation
+################################################################################
+
+#
+# Read a json string from stdin and return the value of the key provided as first argument
+# Uses head -n 1 to avoid duplicates
+#
+function get_json_value () {
+    sed -n 's/.*"'"$*"'": *"\([^"]*\)".*/\1/p' | head -n 1
+}
+
+#
+# Configures variables based on a the docker-machine inspect from $DOCKER_MACHINE_NAME
+#
+function configure_docker_machine {
+  DOCKER_HOST_NAME=$DOCKER_MACHINE_NAME
+  DOCKER_MACHINE_INSPECT=$(docker-machine inspect $DOCKER_MACHINE_NAME)
+  DOCKER_HOST_USER=$(echo -e "$DOCKER_MACHINE_INSPECT" | get_json_value "SSHUser")
+  DOCKER_HOST_IP=$(echo -e "$DOCKER_MACHINE_INSPECT" | get_json_value "IPAddress")
+  DOCKER_MACHINE_STORE_PATH=$(echo -e "$DOCKER_MACHINE_INSPECT" | get_json_value "StorePath")
+
+  DOCKER_HOST_SSH_URL="$DOCKER_HOST_USER@$DOCKER_HOST_IP"
+  DOCKER_HOST_SSH_KEY="$DOCKER_MACHINE_STORE_PATH/id_rsa"
+  DOCKER_HOST_SSH_COMMAND="docker-machine ssh $DOCKER_MACHINE_NAME"
+  log_debug "Running for the docker-machine name: $DOCKER_MACHINE_NAME"
+}
+
+#
+# Initializes and starts up the docker-machine VM.
+#
+function init_docker_machine {
+  log_info "Initializing docker machine $DOCKER_MACHINE_NAME"
+  docker-machine start "$DOCKER_MACHINE_NAME"
+  eval $(docker-machine env "$DOCKER_MACHINE_NAME")
+  configure_docker_machine
+}
+
+################################################################################
+# Generic docker host manipulation
+################################################################################
+
+#
+# Initializes a docker host
+# Initializes boot2docker, unless DOCKER_MACHINE_NAME is defined
+#
+function init_docker_host {
+  if [ -n "$DOCKER_MACHINE_NAME" ]; then
+    init_docker_machine
+  else
+    init_boot2docker
+  fi
 }
 
 #
 # Installs rsync on the Boot2Docker VM, unless it's already installed.
 #
-function install_rsync_on_boot2docker {
-  log_info "Installing rsync in the Boot2Docker image"
+function install_rsync_on_docker_host {
+  log_info "Installing rsync in the Docker Host image"
 
-  boot2docker ssh "if ! type rsync > /dev/null 2>&1; then tce-load -wi rsync; fi"
+  $DOCKER_HOST_SSH_COMMAND "if ! type rsync > /dev/null 2>&1; then tce-load -wi rsync; fi"
```

Comment:
Quick sanity check, in case you haven't done it already: search the code for `boot2docker ssh` calls and make sure all of them are updated to `$DOCKER_HOST_SSH_COMMAND`.